### PR TITLE
experimental: Abort controller

### PR DESCRIPTION
**Pull request by kof**
## Description

I tried how the code will look like with abort controller, if you don't like it - we can close this without merging

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: apps/builder/app/shared/copy-paste/init-copy-paste.ts [hunk 1]
```diff
@@ -1,141 +1,12 @@
...
-  // Capture is required so we get the element before content-editable removes it
-  // This way we can detect when we are inside content-editable and ignore the event
-  document.addEventListener("paste", handlePaste, { capture: true });
-
-  return () => {
-    document.removeEventListener("copy", handleCopy);
-    document.removeEventListener("cut", handleCut);
-    document.removeEventListener("paste", handlePaste, { capture: true });
-  };
+import { initPlugins } from "./init-plugins";
```
Comment (TrySound) [inline apps/builder/app/shared/copy-paste/init-copy-paste.ts:1]: could be in the same module

File: apps/builder/app/canvas/instance-hovering.ts [hunk 2]
```diff
@@ -11,7 +11,11 @@ import type { InstanceSelector } from "~/shared/tree-utils";
 
 type TimeoutId = undefined | ReturnType<typeof setTimeout>;
 
-export const subscribeInstanceHovering = () => {
+export const subscribeInstanceHovering = ({
+  signal,
+}: {
+  signal: AbortSignal;
+}) => {
```
Comment (TrySound) [inline apps/builder/app/canvas/instance-hovering.ts:18]: I would not mix both abort signal and unsubscribe

File: apps/builder/app/canvas/canvas.tsx [hunk 3]
```diff
@@ -130,21 +131,46 @@ const useElementsTree = (
...
+    return subscribeSelected(debounceEffect);
+  }, [debounceEffect]);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    // We need to initialize this in both canvas and builder,
+    // because the events will fire in either one, depending on where the focus is
+    // @todo we need to forward the events from canvas to builder and avoid importing this
+    // in both places
+    initCopyPaste(abortController);
```
Comment (TrySound) [inline apps/builder/app/canvas/canvas.tsx:159]: why not pass signal explicitly?

File: apps/builder/app/canvas/instance-hovering.ts (hunk 2 above)
Comment (kof) [reply apps/builder/app/canvas/instance-hovering.ts:18]: oversight

File: apps/builder/app/canvas/canvas.tsx (hunk 3 above)
Comment (kof) [reply apps/builder/app/canvas/canvas.tsx:159]: idk, thought its unnecessary typing ....

**Review by TrySound — APPROVED** [review]

**Pull request merged**